Implement mother of dragons

cavnak opened the pull request:
This one was a little trickier but thanks to the clean, well organized codebase not too hard.  Props to Cryogen for the project and its excellent structure

cryogen commented on server/game/cards/attachments/04/motherofdragons.js:
```diff
@@ -0,0 +1,48 @@
+                onAttackersDeclared: (event, challenge) => !this.kneeled && this.controller === challenge.attackingPlayer &&_.any(challenge.attackers, card => {
+                    return card.hasTrait('Dragon');
+                })
+                
```
please remove this whitespace and add a semi colon on the line above

cryogen commented on server/game/cards/attachments/04/motherofdragons.js:
```diff
@@ -0,0 +1,48 @@
+            return false;
+        }
+        return super.canAttach(player, card);
+
```
please remove this whitespace

cryogen commented on server/game/cards/attachments/04/motherofdragons.js:
```diff
@@ -0,0 +1,48 @@
+
+class MotherOfDragons extends DrawCard {
+    setupCardAbilities() {
+        this.reaction({
```
As this is a 'Challenges Action' effect, this would be better implemented as an action so that the player has a menu and can trigger the effect when they want, rather than in reaction to attackers/defenders.  It will also simplify the implementation a bit

cryogen commented:
Thanks, just a couple of minor points.  Apologies for being picky on whitespace but it will set eslint off if not fixed :)

Also, you can thank @ystros for the clean code.  My first iteration, while functional. was pretty horrible!

ystros commented:
Just a note - I've started putting together more documentation. You can read a bit more about how to defined actions here, beyond just looking at other examples in the code base:
https://github.com/cryogen/throneteki/blob/master/docs/implementing-cards.md

cryogen commented on server/game/cards/attachments/04/motherofdragons.js:
```diff
@@ -0,0 +1,41 @@
+    addToChallenge(player) {
+        var challenge = this.game.currentChallenge;
+        if(this.kneeled || !player.cardsInPlay.any(card => {
+            return card.inChallenge && card.hasTrait('Dragon');
```
Happy to merge as-is, but just so you know, this.game.currentChallenge.isParticipating(card) is a more generic way to do this

cavnak commented:
No problem, refactored as an action taking into account your feedback.  I am a little confused because I did run it through eslint using the config in the root and it complained about a couple other things I fixed but not that.    Might have something to do with installing it globally.  Should be good now.

EDIT: And argh I was looking for just that and somehow missed it.

cryogen commented:
It might not have picked up the spacing, but it should have picked up the missing semi, regardless, thanks :)
